### DISCUSSION TITLE
Tasmota suitable for controlling custom "multifunctional" board based on ESP32?

### DISCUSSION THREAD

**Discussion by klaus-liebler** [arendst/Tasmota#11356] [Q&A] [answered]
Hi community,

Program Version | 9.3.1(tasmota)
2021-03-09T16:17:11
1_0_5_RC6/v3.3.4-402-gcd59d107b

I developed a custom electronic board to control the awnings and lights of my terrace roofing. It has

- 6 relays (2x2 for the drives of the awnings
- 1 constant current LED driver (PWM)
- 5 POWER-Mosfet to control a RGBCW LED strip
- alternative WS2812
- 433MHz receiver
- If you are interested in details: https://github.com/klaus-liebler/sensact/tree/master/pcb/sensactOutdoor

My very basic question is: Is tasmota the right software to control this board with a smartphone (WebUI) and a 433MHz remote control?

The hardware is not yet on my desk, but I tried to emulate the general behavior with ESP32 DevKitC and some jumper wires :-). After some initial experiments with the template/module configuration it seem as if tasmota makes a close look at the pin settings and makes assumptions on what the device is doing. Unfortunately, my experiments show, that only one type of device functionality is assumed. For example, when I configure a WS2812 output and an additional PWM output , the WEB UI offers a "Select-Color" UI for the WS2812 only. There is no UI element for the additional PWM output. When I remove the WS2812, then the PWM appears.
Is this behaviour by design?  Can I somehow influece, how tasmota maps pin definitions to functionality and how functionality is displayed on the WebUI?

Maybe tasmota is just not designed for an application like this?

Thank you for your support!

Klaus

**Comment by barbudor** [comment, marked as answer]
Hi
Tasmota can do a lot of things but there is quite a path before seeing everything.

When it comes to Lights, I suggest you read https://tasmota.github.io/docs/Lights/ to understand how Tasmota in handling them.
Briefly the original concept was that Tasmota can control 1 light which could be made from up to 5 channels (R, G, B Cold W, Warm W).
This is why when you configure up to 5 PWM, you see such GUI
However there is a SetOption command (one of a many as you will learn) which allows splitting those as 5 Independant channels/
See : https://tasmota.github.io/docs/Commands/#setoption68

Enjoy Tasmota
